examples/sampler.cpp

```diff
@@ -0,0 +1,405 @@
+
+    for (int i = 0; i < weight.Size(); ++i)
+    {
+        // weight[i] = 1.0 / weight[i];
+        weight[i] = 1.0;
+    }
+
+    mfem::L2_FECollection ufec(0, nDimensions);
+    mfem::ParFiniteElementSpace ufespace(pmesh, &ufec);
+
+    // Construct vertex_edge table in mfem::SparseMatrix format
+    auto& vertex_edge_table = nDimensions == 2 ? pmesh->ElementToEdgeTable()
+                              : pmesh->ElementToFaceTable();
+    mfem::SparseMatrix vertex_edge = TableToMatrix(vertex_edge_table);
+
+    // Construct agglomerated topology based on METIS or Cartesian agglomeration
+    mfem::Array<int> partitioning;
+    if (metis_agglomeration)
+    {
+        FESpaceMetisPartition(partitioning, sigmafespace, ufespace, coarseningFactor);
+    }
+    else
+    {
+        auto num_procs_xyz = spe10problem.GetNumProcsXYZ();
+        FVMeshCartesianPartition(partitioning, num_procs_xyz, *pmesh, coarseningFactor);
+    }
+
+    const auto& edge_d_td(sigmafespace.Dof_TrueDof_Matrix());
+
+    auto edge_boundary_att = GenerateBoundaryAttributeTable(pmesh);
+
+    mfem::SparseMatrix W_block = SparseIdentity(vertex_edge.Height());
+
+    const double cell_volume = spe10problem.CellVolume(nDimensions);
+    W_block *= cell_volume * kappa * kappa;
+
+    mfem::Vector mean_fine(ufespace.GetVSize());
+    mean_fine = 0.0;
+    mfem::Vector mean_upscaled(ufespace.GetVSize());
+    mean_upscaled = 0.0;
+    mfem::Vector m2_fine(ufespace.GetVSize());
+    m2_fine = 0.0;
+    mfem::Vector m2_upscaled(ufespace.GetVSize());
+    m2_upscaled = 0.0;
+
+    int total_coarse_iterations = 0;
+    int total_fine_iterations = 0;
+    double total_coarse_time = 0.0;
+    double total_fine_time = 0.0;
+
+    // Create Upscaler
+    auto fvupscale = std::make_shared<FiniteVolumeUpscale>(
+                         comm, vertex_edge, weight, W_block, partitioning, *edge_d_td,
+                         edge_boundary_att, ess_attr, spect_tol, max_evects, dual_target, scaled_dual,
+                         energy_dual, hybridization);
+
+    fvupscale->MakeFineSolver();
+    fvupscale->PrintInfo();
+    fvupscale->ShowSetupTime();
+
+    const int num_aggs = partitioning.Max() + 1;
+    if (myid == 0)
+        std::cout << "Number of aggregates: " << num_aggs << std::endl;
+    PDESampler pdesampler(fvupscale, ufespace.GetVSize(), num_aggs, nDimensions,
+                          cell_volume, kappa, seed + myid);
+
+    double max_p_error = 0.0;
+    for (int sample = 0; sample < num_samples; ++sample)
+    {
+        double count = static_cast<double>(sample) + 1.0;
+        pdesampler.NewSample();
+
+        auto sol_coarse = pdesampler.GetCoarseCoefficientForVisualization();
+        auto sol_upscaled = fvupscale->Interpolate(sol_coarse);
+        for (int i = 0; i < sol_upscaled.Size(); ++i)
+            sol_upscaled(i) = std::log(sol_upscaled(i));
+        fvupscale->Orthogonalize(sol_upscaled); // can we orthogonalize on coarse grid?
```

Comment:
I think you can do so once you have the coarse representation of constant 1 (which you have made it available in this PR). 

In fine level this is what we do:
vec -= (1^T * vec) 1

Let P be the interpolation matrix. If we have a coarse solution vec_c, and a coarse representation of fine constant 1 (denoted by 1_c = P 1), then of course we can interpolate all the coarse quantities to fine level and use the fine level orthogonalization (and this is what the above few lines is doing):
P vec_c -= ( (P 1_c)^T * (P vec_c) ) P 1_c

The above formula can be simplified because our P satisfies P^T P = I. So 
( (P 1_c)^T * (P vec_c) ) P 1_c = ( (1_c)^T P^T * P vec_c) P 1_c = P ( (1_c)^T * vec_c) 1_c

So the coarse level orthogonalization is 
vec_c -= ( (1_c)^T *  vec_c) 1_c

You can interpolate to fine level afterwards if you want.